sys/reaper/reaper_unix.go

```diff
@@ -33,35 +33,58 @@ var ErrNoSuchProcess = errors.New("no such process")
 
 const bufferSize = 2048
 
+type subscriber struct {
+	sync.Mutex
+	c      chan runc.Exit
+	closed bool
+}
+
+func (s *subscriber) close() {
+	s.Lock()
+	if s.closed {
+		return
```

Comment:
Is it ok to leave the subscriber locked here?

Reply:
it is not safe. we should unlock it before return